libraries/AP_NavEKF2/AP_NavEKF2_core.cpp

```diff
@@ -1529,57 +1529,57 @@ void NavEKF2_core::MagTableConstrain(void)
 void NavEKF2_core::ConstrainStates()
 {
     // attitude errors are limited between +-1
-    for (uint8_t i=0; i<=2; i++) statesArray[i] = constrain_float(statesArray[i],-1.0f,1.0f);
+    for (uint8_t i=0; i<=2; i++) statesArray[i] = constrain_ftype(statesArray[i],-1.0f,1.0f);
     // velocity limit 500 m/sec (could set this based on some multiple of max airspeed * EAS2TAS)
-    for (uint8_t i=3; i<=5; i++) statesArray[i] = constrain_float(statesArray[i],-5.0e2f,5.0e2f);
-    // position limit 1000 km - TODO apply circular limit
-    for (uint8_t i=6; i<=7; i++) statesArray[i] = constrain_float(statesArray[i],-1.0e6f,1.0e6f);
+    for (uint8_t i=3; i<=5; i++) statesArray[i] = constrain_ftype(statesArray[i],-5.0e2f,5.0e2f);
+    // position limit - TODO apply circular limit
+    for (uint8_t i=6; i<=7; i++) statesArray[i] = constrain_ftype(statesArray[i],-EK2_POSXY_STATE_LIMIT,EK2_POSXY_STATE_LIMIT);
```

Comment:
State limit for double precision has been opened up to 50000 km which may break at some point on a  round earth trip, but is less evil than hitting a hard limit of 1000km so OK for now.

Reply:
yep. i have a separate patch that adds a wrapping function for the position Vector3F. I need to discuss that with you though and thought it was out of scope for this PR

Reply:
Agree.